server/routes/user.js

```diff
@@ -0,0 +1,160 @@
+import { Router } from 'express'
+import ensureAuthenticated from '../utils/passport/ensure-authenticated'
+import { routerRoutes } from '../utils/routes'
+
+const router = Router()
+
+router
+  .route(routerRoutes.userSiteSubjects)
+  .get(ensureAuthenticated, async (req, res) => {
+    try {
+      const { prisma } = req.app.locals
+      const { siteSubjects } = await prisma.users.findUnique({
+        where: { uid: req.params.uid },
+        select: { siteSubjects: true },
+      })
+
+      return res.status(200).json({ data: siteSubjects })
+    } catch (error) {
+      console.error(error.stack)
+
+      return res.status(500).send({ message: error.message })
+    }
+  })
+
+router
+  .route(routerRoutes.userSiteSubjectsStar)
+  .post(ensureAuthenticated, async (req, res) => {
+    try {
+      const { prisma } = req.app.locals
+      const user = await prisma.users.findUnique({
+        where: { uid: req.params.uid },
+        select: {
+          siteSubjects: true,
+        },
+      })
+      const { siteSubjects } = user
+      const siteSubjectSiteIndex = siteSubjects.findIndex(
+        (siteSubject) => siteSubject.site === req.body.site
+      )
+      if (siteSubjectSiteIndex > -1) {
+        siteSubjects[siteSubjectSiteIndex].starredSubjects.push(
+          req.body.subject
+        )
+      } else {
+        siteSubjects.push({
+          site: req.body.site,
+          starredSubjects: [req.body.subject],
+        })
+      }
```

Comment:
can we extract this logic to a helper function that can be tested?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/237/commits/51b526b1e0de6c3d30b832a596eb04a45440735e